### ISSUE TITLE
MachineKey instructions for PowerBI ReportServer

### ISSUE THREAD

**Issue by gregoriogreen** [Microsoft/Reporting-Services#49]
The instructions for adding the MachineKey reference items that do not exist in the PowerBI Report Server installation. 

It seems possible, if unlikely, that the user should add the MachineKey to the ReportServer install AND to the applicable or available files in the PowerBI ReportServer Install. 

It may be that PowerBI ReportServer Install differences should be highlighted in the instructions in some way.

Thank you,

**Comment by jtarquino** [comment]
Can you elaborate a little bit more about what change you expect in the documentation?
Also feel free to send a Pull Request with the change you would like to see there

**Comment by gregoriogreen** [comment]
I am sitting here working with the MachineKeys for the PowerBI ReportServer (PBRS). I suspect very soon everyone will be deploying PBRS and not SRS ReportServer. So the documentation should reflect the user demand by highlighting PBRS differences.

The MachineKey section of the instructions does not highlight what to do with MachineKeys for PBRS. Also each time I add a MachineKey to web.config then restart PBRS the MachineKey is stripped out of web.config. So it's a mystery to me.

Are there interdependencies between the SRS install and PBRS install, or is PBRS totally stand-alone?

**Comment by jtarquino** [comment]
The sample explains what should be done for PBIRS and for the next version of SSRS https://github.com/Microsoft/Reporting-Services/tree/master/CustomSecuritySample , for 2016 is a specific sample https://github.com/Microsoft/Reporting-Services/tree/master/CustomSecuritySample2016, those are different as the title shows , I agree we should update the title in the one for PBIRS and SSRS vNext to reflect it doesn't apply to SSRS 2016.

If you look closely the machineKey instructions are different , in PBIRS and for SSRS vNext you don't touch the web.config, you add it to the centralized file where our configuration is the RSReportServer.config so you don't have to replicate it in multiple places
https://github.com/Microsoft/Reporting-Services/tree/master/CustomSecuritySample#step-4-generate-machine-keys

For your last question PBIRS is totally stand alone

**Comment by gregoriogreen** [comment]
I was able to resolve this. The root cause of the issue was a misinterpretation of the instructions. As I am running SQL2016 I tried to use the CustomSample2016 while following the ReportServer PBI instructions wherever I could find them (in CustomSample).

It might be helpful to have three versions of instruction

- SRS Prior to SQL2016
- SRS 2016
- PBI [version] which deserves a stand-alone sample and instruction.

Thank you, this is great software. Looking forward to the OAuth2 Azure AD / Office 365 Auth Sample.

**Comment by danielnord** [comment]
To help other developers that had the same problem this were the steps I took to make it work after I had set the wrong MachineKeys:
1. Generate machine keys AES/AES (Generate from IIS > Site > Machine Key > Generate keys)
2. Set the keys in config file as described in solution
3. Open **Report Server Configuration Manager**
4. Go to **Encryption keys**
5. **Delete** Encrypted content
6. Verify that the Database has credentials
